Export GRAPHQL_MAX_INT and GRAPHQL_MIN_INT

tofran opened the pull request:
Closes #2524

(Replaces abandoned #2582)

IvanGoncharov commented on src/type/scalars.ts:
```diff
@@ -14,8 +14,8 @@ import { GraphQLScalarType } from './definition';
 // they are internally represented as IEEE 754 doubles.
-const MAX_INT = 2147483647;
-const MIN_INT = -2147483648;
+export const MAX_INT = 2147483647;
```
Please convert above comment to JSDoc
If you have time please write a separate docs for both min and max

IvanGoncharov replied:
Also since we making it public please rename it to `GRAPHQL_INT_MAX` and `GRAPHQL_INT_MIN`.

jen801 replied:
👍🏻

IvanGoncharov reviewed (changes requested):
Please export it in `src/type/index.ts` and `src/index.ts`.
Also please check tests folder for the places where this number (or +-1) is used and replace them with this constant.

tofran commented:
Hello @IvanGoncharov , thank you for bringing these in and your detailed comment.
I have made the requested changes:

- [x] Checked the tests folder for the places where this number (or +-1) is used and replace them with this constant.*

```sh
$ grep -R "214748364" src  # Intentionally missing the last character
src/type/scalars.ts:export const GRAPHQL_MAX_INT = 2147483647;
src/type/scalars.ts:export const GRAPHQL_MIN_INT = -2147483648;
$ grep -R "2^31" src
src/type/scalars.ts:    'The `Int` scalar type represents non-fractional signed whole numeric values. Int can represent values between -(2^31) and 2^31 - 1.',
$ grep -R "2 \*\* 31" src

```

- [x] Converted above comment to JSDoc - separated and simplified
- [x] Exported it in src/type/index.ts and src/index.ts.
- [x] Renamed to `GRAPHQL_INT_MAX` and `GRAPHQL_INT_MIN`.

---

*Were you also suggesting that tests to these boundaries should be changed?
Ex:
```js
expect(() => parseLiteral('-9876504321')).to.throw(
  'Int cannot represent non 32-bit signed integer value: -9876504321',
);
```
Would now be
```js
expect(() => parseLiteral(GRAPHQL_MIN_INT-1)).to.throw(
  `Int cannot represent non 32-bit signed integer value: ${GRAPHQL_MIN_INT-1}`,
);
```

Should I make this change? Thank you.

IvanGoncharov commented on src/index.ts:
```diff
@@ -61,6 +61,9 @@ export {
   DEFAULT_DEPRECATION_REASON,
+  /** Int boundaries constants */
+  GRAPHQL_MAX_INT,
+  GRAPHQL_MIN_INT,
```
Please match the order in `types/index.ts`

IvanGoncharov replied:
Everything else looks good.
Happy to merge once fixed.

tofran commented:
@IvanGoncharov @saihaj thank you.
Just pushed the requested changes, ready to merge.

tofran commented:
@IvanGoncharov I don't think we will be able to merge until you re-review.
Thanks in advance.

IvanGoncharov commented:
@tofran Thanks 👍 
Merged 🎉

jen801 reviewed (commented):
#3355

saihaj commented:
Just to avoid noise from potential bot accounts.
